include/storage/shared_monitor.hpp

```diff
@@ -126,7 +135,9 @@ template <typename Data> struct SharedMonitor
         }
         catch (const bi::interprocess_exception &exception)
         {
-            return false;
+            auto message =
```

Comment:
This changes the expected behaviour of this function. The exception is an indication of the shared memory object not existing.